src/main/java/org/craftercms/deployer/impl/TargetServiceImpl.java

```diff
@@ -283,6 +286,79 @@ public void recreateIndex(String env, String siteName) throws TargetNotFoundExce
         adminService.recreateIndex(target.getId());
     }
 
+    /**
+     * Duplicates the index from the source site to the target site.
+     *
+     * @param sourceTarget the source site
+     * @param siteName     the target site name
+     */
+    private void duplicateIndex(final Target sourceTarget, final String siteName) {
+        String indexIdFormat = sourceTarget.getConfiguration().getString("target.search.indexIdFormat");
+        ApplicationContext appContext = sourceTarget.getApplicationContext();
+        OpenSearchAdminService adminService = appContext.getBean(OpenSearchAdminService.class);
+        adminService.duplicateIndex(format(indexIdFormat, sourceTarget.getSiteName()), format(indexIdFormat, siteName));
+    }
+
+    @Override
+    public void duplicateTarget(final String env, final String sourceSiteName, final String siteName)
+            throws TargetNotFoundException, TargetAlreadyExistsException, TargetServiceException {
+        Target sourceTarget = getTarget(env, sourceSiteName);
+        try {
+            // Create new target file from the source target
+            duplicateTargetConfigurations(sourceTarget, siteName);
```

Comment:
What happens if there is a partial failure? For example, what if the search engine was not available?

Reply:
Then partially created target will remain. I will fix